<?xml version="1.0" ?>
<launch>

<arg name="camera" default="/front"/>
<arg name="min_thres_defects" default="0.005"/>
<arg name="max_thres_defects" default="0.04"/>
<arg name="global_align_thres" default="0.02"/>
<arg name="generate_cloud" default="true"/>
<!-- Point cloud generation -->
  <node name="uncompresser" pkg="image_transport" type="republish" if="$(arg generate_cloud)"
        args="compressedDepth in:=/front_web/depth_registered/image_raw raw out:=/front/depth_registered/image_raw"/>

 <node pkg="depth2cloud" type="depth2cloud_node" name="depth2cloud_node"  if="$(arg generate_cloud)">
    <remap from="depth" to="/front/depth_registered/image_raw"/>              # Topic for the camera depth
    <remap from="info" to="/front_web/depth_registered/camera_info"/>   # Topic for the camera info
    <remap from="cloud" to="/front/points"/>                # Topic name for output point cloud
    <param name="range_min" value = "0.7"/>                             # Min range in meters for point cloud generation
    <param name="range_max" value = "6.0"/>                             # Max range in meters for point cloud generation
    <param name="down_sampling" value = "1"/>                   # Camera down sampling factor. Values: 1, 2 or 4. The bigger the faster.
  </node>

<!-- Point Cloud Processing node -->
<node name="test_analysis" pkg="pcl_processing" type="map_analysis" output="screen">
  <remap from="points2" to="$(arg camera)/points"/>
  <param name="min_thres_defects" value="$(arg min_thres_defects)"/>
  <param name="max_thres_defects" value="$(arg max_thres_defects)"/>
  <param name="global_align_thres" value="$(arg global_align_thres)"/>
</node>

</launch>
